backend/editor/entries.py

```diff
@@ -160,6 +149,25 @@ async def get_and_parse_taxonomy(self, uploadfile: UploadFile | None = None):
             log.exception(e)
             raise e
 
+    async def fetch_external_taxonomy_files(self, tmpdir: str) -> list[str]:
+        """
+        Helper function to fetch external taxonomies concurrently from Github
+        """
+        external_taxonomy_filepaths = []
+        tasks = []
+
+        # Create tasks for each external taxonomy and store them in a list
+        for external_taxonomy in EXTERNAL_TAXONOMIES[self.taxonomy_name]:
+            task = asyncio.create_task(self.get_github_taxonomy_file(tmpdir, external_taxonomy))
+            tasks.append(task)
+
+        # Wait for all tasks to complete concurrently
+        for task in tasks:
+            external_filepath = await task
```

Comment:
big nit (feel free to ignore): isn't using something like `asyncio.gather` or `asyncio.TaskGroup` better here?
Also I believe you could use `anyio` directly which the concurrency library used by FastAPI under the hood  https://fastapi.tiangolo.com/async/#write-your-own-async-code